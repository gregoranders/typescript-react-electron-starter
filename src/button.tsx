import * as React from "react";

export interface IButtonProperties {
  label: string;
  onClick: () => void;
}

export interface IButtonState {
  toggle: boolean;
}

export class Button extends React.Component<IButtonProperties, IButtonState> {

  public constructor(props: IButtonProperties) {
    super(props);
    this.state = {
      toggle: false,
    };
  }

  public render(): JSX.Element {
    return (<div>
      <h2>{ !this.state.toggle ? "1" : "2"}</h2>
      <button className="btn btn-primary"
      onClick={ (event: React.MouseEvent<HTMLButtonElement>): void => { this.handleClick(event); } }>
        {this.props.label}
      </button>
    </div>);
  }

  protected handleClick(event: React.MouseEvent<HTMLButtonElement>): void {
    this.props.onClick();
    this.setState({
      toggle: !this.state.toggle,
    });
  }
}
