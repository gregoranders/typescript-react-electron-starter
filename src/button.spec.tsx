import { mount, ReactWrapper } from "enzyme";
// import * as Adapter from "enzyme-adapter-react-16";
import * as React from "react";
import * as renderer from "react-test-renderer";

import * as TestSubject from "./button";

describe("Button", (): void => {

  // beforeAll((): void => {
  //   configure({ adapter: new Adapter() });
  // });

  it("Button export exists", (): void => {
    expect(TestSubject.Button).toBeDefined();
  });

  it("Button snapshot", (): void => {
    const snapshot: renderer.ReactTestRenderer
      = renderer.create(<TestSubject.Button label="button" onClick={ (): void => { return; } } />);
    snapshot.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it("Button clicked mount", (): void => {
    let counter: number = 0;
    const func: jest.Mock<void, any[]> = jest.fn( (): void => {
      counter = counter + 1;
    });
    const testSubject: ReactWrapper<TestSubject.IButtonProperties, TestSubject.IButtonState, TestSubject.Button>
      = mount(<TestSubject.Button label="button" onClick={ func } />);

    expect(testSubject.text()).toBe("1button");
    expect(counter).toBe(0);
    expect(testSubject.state().toggle).toBeFalsy();
    expect(func.mock.calls.length).toBe(0);

    testSubject.find("button").simulate("click");

    expect(testSubject.text()).toBe("2button");
    expect(counter).toBe(1);
    expect(testSubject.state().toggle).toBeTruthy();
    expect(func.mock.calls.length).toBe(1);
  });
});
