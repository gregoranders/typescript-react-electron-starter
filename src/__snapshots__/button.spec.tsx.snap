// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button snapshot 1`] = `
<div>
  <h2>
    1
  </h2>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    button
  </button>
</div>
`;
